### PR TITLE
Modify torch.movedim to handle scalar as no-op

### LINKED ISSUE #69432: `torch.movedim` trigger internal assert failure!
## 🐛 Bug

When the tensor is 0 dimensional and both source and destination are 0, `torch.movedim` will trigger internal assert failure.

## To Reproduce

```python
import torch

tensor = torch.rand(torch.Size([]))
res1 = torch.movedim(tensor, 0, 0)
# RuntimeError: std::distance(source_dims.begin(), source_iter) == rest_dimINTERNAL ASSERT FAILED at "../aten/src/ATen/native/TensorShape.cpp":2448, please report a bug to PyTorch.
```

## Expected behavior

Do not trigger internal assert failure.

## Environment

PyTorch Version (e.g., 1.0): 1.10.0
OS (e.g., Linux): Linux
How you installed PyTorch (conda, pip, source): conda
Python version: 3.9.5
CUDA/cuDNN version: 11.1
GPU models and configuration: RTX 3090

cc @brianjo @mruberry

### PR DESCRIPTION
**Pull request by francescocastelli**
Summary: `torch.movedim` directly handle the case of a scalar tensor (0-dim) in input as a no-op by returning a view of the input tensor (after all the usual checks for the other parameters)

Test:  This code now works fine and res1 is a view of tensor
```
import torch

tensor = torch.rand(torch.Size([]))
res1 = torch.movedim(tensor, 0, 0)
```

Fixes #69432

### CODE REVIEW

**Comment by vadimkantorov** [comment]
Should it be a copy or just a reference? The regular movedim doesn't clone the input tensor, right, just returns a view?

**Comment by francescocastelli** [comment]
Yes it returns a view, I wrongly written in the description 'copy'. I have changed now so it's more clear

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -2447,6 +2447,10 @@ Tensor movedim(const Tensor& self, IntArrayRef src, IntArrayRef dst) {
   TORCH_CHECK(all_unique(normalized_src), "movedim: repeated dim in `source` (", src, ")");
   TORCH_CHECK(all_unique(normalized_dst), "movedim: repeated dim in `destination` (", dst, ")");
 
+  // handle the case of scalar tensor as a no-op
+  if (self_dim == 0)
+    return self.view(self.sizes());
```
Comment (albanD) [inline aten/src/ATen/native/TensorShape.cpp:2452]: nit: `self.alias()` would be simpler here.

Comment (francescocastelli) [reply aten/src/ATen/native/TensorShape.cpp]: rigth, done!

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by YangChenyuan** [comment]
@albanD Can you see my reply about why this API should raise an error for tensor without dimension? https://github.com/pytorch/pytorch/issues/69432#issuecomment-988803499

**Comment by ilovepytorch** [comment]
@albanD @francescocastelli Can you see my reply about why this kind of API should raise an error for tensor without dimension? https://github.com/pytorch/pytorch/issues/69432#issuecomment-988803499

**Comment by francescocastelli** [comment]
@ilovepytorch Sorry I didn't see that, ok for me. I'll wait for @albanD to reply

**Comment by ilovepytorch** [comment]
> @ilovepytorch Sorry I didn't see that, ok for me. I'll wait for @albanD to reply

Thanks!

**Comment by albanD** [comment]
No-op is still better than an internal assert :D
But yes we should keep discussing no-op vs nice error message.

**Pull request closed without merging**